### PR TITLE
Use parse only if not an icon object

### PR DESCRIPTION
**Pull request by robmadole**

### CODE REVIEW

File: src/components/__tests__/FontAwesomeIcon.test.js [hunk 1]
```diff
@@ -33,7 +37,7 @@ if(coreHasFeature(ICON_ALIASES)) {
     const vm = mountFromProps({ icon: ['fas', 'close'] })
 
     expect(vm.$el.tagName).toBe('svg')
-    expect(vm.$el.classList.contains('fa-close')).toBeTruthy()
+    expect(vm.$el.classList.contains('fa-xmark')).toBeTruthy()
```
Comment (jasonlundien) [inline src/components/__tests__/FontAwesomeIcon.test.js:40]: did you want the `fa-xmark` icon here?  
In v6 `xmark` is the name of the icon and `close` is the alias.
If we keep the `fa-mark` we will need to import it (line 4), and will probably want to change the test name; currently test is 'find a free-solid-svg-icon that is an alias'.
As of right now, the test mentioned on the previous line fails because `xmark` has NOT been imported.

File: src/components/FontAwesomeIcon.js [hunk 2]
```diff
@@ -12,7 +18,7 @@ function normalizeIconArgs (icon) {
     return null
   }
 
-  if (typeof icon === 'object' && icon.prefix && icon.iconName) {
+  if (icon && typeof icon === 'object' && icon.prefix && icon.iconName) {
```
Comment (jasonlundien) [inline src/components/FontAwesomeIcon.js:21]: I see that `icon.icon` is the only difference here between current line 21 and line 9 change. 
Would we ever not have an `icon.icon` property?
I see that this property is the icon details, unicode, svg, etc.  My only thought here is that this covers webfonts in the case of an icon not using an svg ... is this correct, so we want to return the icon even if NOT an svg ?

**Review by jasonlundien — COMMENTED** [review]
A couple of questions for clarification's sake on my part.

**Comment by jasonlundien** [comment]
PR #327 is no longer needed — all its relevant changes are already in the codebase (or no longer apply).

Here's the breakdown:

**Code change** (the core of the PR): The early return for icon objects that have `.prefix`, `.iconName`, AND `.icon` (i.e., icon objects imported directly from icon packages without being added to the library) is already in `FontAwesomeIcon.js:9`:

```js
if (icon && typeof icon === 'object' && icon.prefix && icon.iconName && icon.icon) {
  return icon
}
```

**Test coverage**: A test for this exact behavior exists at `FontAwesomeIcon.test.js:459-464` — `'using imported object from svg icons package'` with `mountFromProps({ icon: faUser })`.

**`package.json` test scripts**: The PR swapped `test.next` for `test.next.proregistry` in the `test` script. This is moot — the project has since moved from Jest to Vitest and those multi-version test scripts no longer exist.

**Minor note**: The PR also added an `icon &&` null guard to line 21 (`if (typeof icon === 'object' && icon.prefix && icon.iconName)`). The current code is missing that guard, but it's functionally a non-issue because `icon === null` is already caught on line 17 before reaching line 21 — only `undefined` could reach it, and `undefined && ...` would have been `false` anyway in the original. So no practical difference.

**Bottom line**: This PR can be closed as already resolved.
